### PR TITLE
[#9382] Add auto-sync of front-end API request format with back-end

### PR DESCRIPTION
**Pull request by xpdavid**
Part of #9382

The first two commits are just to migrate the frontend to use the new auto-sync feature introduced in `#9079` for API output.

### CODE REVIEW

File: static-analysis/teammates-macker.xml [hunk 1]
```diff
@@ -91,6 +91,7 @@
                     <deny>
                         <from class="${ui}.**" >
                             <exclude class="${ui}.webapi.output.**" /> <!-- allow dependency within the package -->
+                            <exclude class="${ui}.webapi.request.**" />
```
Comment (wkurniawan07) [inline static-analysis/teammates-macker.xml]: I think we need an entirely new section for `ui.webapi.request` package.

File: build.gradle [hunk 2]
```diff
@@ -124,19 +124,38 @@ tasks.withType(cz.habarta.typescript.generator.gradle.GenerateTask) {
...
+    ]
+    mapEnum = "asEnum"
+    nonConstEnums = true
 }
 
 task generateApiOutputFormat(type: cz.habarta.typescript.generator.gradle.GenerateTask) {
     classPatterns = [
-        "teammates.ui.webapi.output.**"
+        "teammates.ui.webapi.output.**",
+        "teammates.common.datatransfer.questions.**"
```
[the remark this answers is not in the document]
Comment (xpdavid) [reply build.gradle]: `teammates.ui.webapi.output.**` can only link to the abstract class `Feedback(Question|Response)Detail`. To get the concrete implementation (e.g. FeedbackTextQuestionDetail), we need this.

**Comment by wkurniawan07** [comment]
@xpdavid For this PR we'll also handle this suppression:
https://github.com/TEAMMATES/teammates/blob/b03ac67d935a459ed7f90aad1640f197beb29300/static-analysis/teammates-checkstyle-suppressions.xml#L68

**Comment by xpdavid** [comment]
> For this PR we'll also handle this suppression:

On top of the PR, violations found only in `GetInstructorCourseDetailsAction` and `GetStudentCoursesAction` which are not in the scope of the PR. The suppression should be removed after the API consolidation.

**Pull request merged**